lib/identity_cache.rb

```diff
@@ -191,6 +193,21 @@ def fetch_multi(*keys)
       result
     end
 
+    def with_deferred_parent_expiration
+      raise NestedDeferredParentBlockError if Thread.current[:deferred_parent_expiration]
+
+      Thread.current[:deferred_parent_expiration] = true
+      Thread.current[:parent_records_for_cache_expiry] = Set.new
+
+      yield
+
+      Thread.current[:deferred_parent_expiration] = nil
+      Thread.current[:parent_records_for_cache_expiry].each(&:expire_primary_index)
+    ensure
+      Thread.current[:deferred_parent_expiration] = nil
+      Thread.current[:parent_records_for_cache_expiry].clear
```

Comment:
I agree we just want to empty this vs trying to process it in the ensure block... all real processing we expect to be handled normally in the around yield this cleans up in both standard and exceptional cases